src/conn_handler.c

```diff
@@ -444,7 +451,9 @@ static int handle_ascii_client_connect(statsite_conn_handler *handle) {
 
         // Convert the value to a double
         val = str2double(val_str, &endptr);
-        if (unlikely(endptr == val_str)) {
+        if (unlikely(endptr == val_str ||
+              (errno == ERANGE &&
+                 (val == HUGE_VAL || val == -HUGE_VAL || val == 0)))) {
```

Comment:
I don't think we even need to check `val` if `errno == ERANGE` that is enough to error out.